packages/web/src/hooks/usePropagateEvent.ts

```
@@ -0,0 +1,39 @@
+
+interface CustomPropagateEvent extends Event {
+  arguments: unknown[];
+}
+
+export default function usePropagateEvent({
+  pushStateCallbackFn = () => {},
+  replaceStateCallbackFn = () => {},
+}: UsePropagateEventCallbackFn) {
+  useEffect(() => {
+    var _wr = function (type: keyof History) {
+      var orig = history[type] as Function;
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)